[5.5] Use of argument (un)packing in throw_if/unless helpers

kduma opened the pull request:
This change allows to pass more than one argument to the exception class.

For example [base exception](http://php.net/manual/en/exception.construct.php) class accepts message and code, so after this PR we can do this:
```php
throw_if($is_error, \Exception::class, $message, $code);
```

kduma committed:
Use of argument (un)packing in throw_if/unless helpers

JosephSilber commented on src/Illuminate/Support/helpers.php:
```diff
@@ -893,13 +893,13 @@ function tap($value, $callback = null)
      * @return void
      */
-    function throw_if($boolean, $exception, $message = '')
+    function throw_if($boolean, $exception, ...$arguments)
```
I would rename this to `$parameters`.

kduma replied:
True. At least I've learned [something](http://stackoverflow.com/questions/1788923/parameter-vs-argument) :)

JosephSilber replied:
Uh... I've got that backwards.

I was simply saying that I'd use `$parameters` because that's what we use all over in our `__call` methods. I wasn't thinking from a semantic perspective.

From a semantic perspective, `$arguments` actually makes _more_ sense.

So I take back what I said originally. I now believe that `$arguments` is actually a better name :see_no_evil:

kduma committed:
Replace $arguments with $parameters

because "argument is the value/variable/reference being passed in, parameter is the receiving variable used w/in the function/block" :)
